hub/collections/dataset/__init__.py

```diff
@@ -132,3 +148,54 @@ def merge(datasets: Iterable[Dataset]) -> Dataset:
             assert tname not in tensors
             tensors[tname] = tvalue
     return Dataset(tensors)
+
+# ItemClass to Dtags mapping
+dtags = {
+    'Audio': 'audio',
+    'BBoxFeature': 'box',
+    'Image': 'image',
+    'Dtype': 'tensor',
+    'Tensor': 'tensor',
+    'ClassLabel': 'text',
+    'Text': 'text',
+    'Video': 'video',
+#    'Sequence': 'object',
+#    'FeaturesDict': 'object',
+}
+
+# Dtag to Archiver mapping
+codecs = {
+    'audio': 'lz4:4',
+    'box': 'lz4:4',
+    'image': 'lz4:4',
+    'tensor': 'lz4:4',
+    'text': 'lz4:4',
+    'video': 'lz4:4',
+}
+
+
+def from_tensorflow(ds_tf, info, n_examples=-1):
+    #0. Auxilary variables
+    tensors = {feature: [] for feature in info.features.keys()}
+    #1. Subsampling the input dataset for testing issues
+    if n_examples > 0:
+        ds_tf = ds_tf.take(n_examples)
+    #2. Mapping features to collected tensors from the input dataset
+    for ex in ds_tf:
+        for feature, tensor in ex.items():
+            if feature in dtags:
+                tensors[feature] += [tensor]
+    #3. Mapping features to concatenated tensors
+    for feature in tensors:
+        data = np.array(tensors[feature])
+        name = type(info.features[feature]).__name__
+        dtag = dtags.get(name, 'tensor')
+        codec = codecs[dtag]
+        tensors[feature] = hub_tensor.from_array(data, dtag=dtag, dcompress=codec)
```

Comment:
using `.from_array` won't handle large datasets that numpy cannot handle it, either we should switch to data generator or creating tensors in chunked way and then combining them.